mongodbatlas.go

```diff
@@ -180,13 +196,23 @@ func (m *MongoDBAtlas) DeleteUser(ctx context.Context, req dbplugin.DeleteUserRe
 		}
 	}
 
-	// Default to "admin" if no db provided
-	if databaseUser.DatabaseName == "" {
-		databaseUser.DatabaseName = "admin"
+	// If the user is an X.509 user, delete the user from the X.509 certificate subject field
+	if strings.HasPrefix(req.Username, "CN=") {
```

Comment:
nit: is it worth moving this into a constant, something like `x509UserPrefix`? fine to leave it as is if we're only doing the prefix checks in 1 or 2 spots and don't foresee doing any more 😄

Reply:
I'll leave it as is for now since we are only checking it in two spots